content/docs/hooks-recipes.md

```diff
@@ -10,3 +10,157 @@ prev: hooks-faq.html
 This is a collection of recipes that may help you start "thinking in Hooks". Feel free to play with them or copy and paste them into your project or tweak them to your needs.
 
 ## Data Fetching
+
+## Subscriptions
+
+Sometimes React components need to subscribe to an external data source. The `useSubscription` hook below shows how to do this in a way that is compatibile with React's upcoming concurrent rendering mode.
+
+> Note
+>
+> Subscriptions may not be the best solution for your use case. For example:
+> * Redux/Flux stores should use the [context API](/docs/context.html) instead.
+> * Complex libraries like Relay/Apollo should manage subscriptions manually with the same technique demonstrated below.
+> * In the future, I/O subscriptions (e.g. notifications) that update infrequently should use [`react-cache`](https://github.com/facebook/react/blob/master/packages/react-cache/README.md).
+
+Although `useSubscription` guarantees correctness, it achieves this by [sometimes de-opting to synchronous mode](https://github.com/facebook/react/issues/13186#issuecomment-403959161). This is an inherent limitation of storing state outside of React's managed state queue and rendering in response to a change event. For **full compatibility** with concurrent rendering, including both **time-slicing** and **React Suspense**, the suggested longer-term solution is to move to one of the patterns described in the note above.
```

Comment:
People don't know what "de-opting" or "synchronous mode" is here. We didn't introduce these concepts anywhere in the docs except the blog (which isn't a required reading for docs). I'd omit details about any future features here, and just say it is compatible with Strict Mode (which *is* in the docs).